16 Validation/readme_es.md

```diff
@@ -0,0 +1,305 @@
+        onLogin={this.onLogin}
+        onUpdateField={this.onUpdateLoginField}
+        loginInfo={this.state.loginInfo}
++       loginFormErrors={this.state.loginFormErrors} 
+      />
+    )
+  }
+```
+
+- Ahora necesitamos definir la propiedad en el componente loginForm.
+
+_./src/loginForm.tsx_
+
+```diff
+import { LoginEntity } from "../../model/login";
++ import { LoginFormErrors } from './viewmodel';
+
+interface Props {
+  loginInfo: LoginEntity;
+  onUpdateField: (string, any) => void;
+  onLogin: () => void;
++  loginFormErrors : LoginFormErrors;
+}
+```
+
+- Ahora actualiza nuestros componentes para encontrar el componente input nuevo.
+
+_./src/common/pages/loginForm.tsx_
+
+```diff
+import { LoginEntity } from "../../model/login";
+import { LoginFormErrors } from './viewmodel';
++ import { TextFieldForm } from '../../common/forms/textFieldForm';
+```
+
+_./src/common/pages/loginForm.tsx_
+
+```diff
+export const LoginForm = (props: Props) => {
+-   const { onLogin, onUpdateField, loginInfo } = props;
++   const { onLogin, onUpdateField, loginInfo, loginFormErrors } = props;
+
+-  const onTexFieldChange = (fieldId) => (e) => {
+-    onUpdateField(fieldId, e.target.value);
+-  }
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center' }}>
+-      <TextField
+-        label="Name"
+-        margin="normal"
+-        value={loginInfo.login}
+-        onChange={onTexFieldChange('login')}
+-      />
++      <TextFieldForm
++        label="Name"
++        name="login"
++        value={loginInfo.login}
++        onChange={onUpdateField}
++        error={loginFormErrors.login.errorMessage}
++      />
+
+-      <TextField
+-        label="Password"
+-        type="password"
+-        margin="normal"
+-        value={loginInfo.password}
+-        onChange={onTexFieldChange('password')}
+-      />
++      <TextFieldForm
++        label="Password"
++        name="password"
++        value={loginInfo.password}
++        onChange={onUpdateField}
++        error={loginFormErrors.password.errorMessage}
++      />
+
+
+      <Button variant="contained" color="primary" onClick={onLogin}>
+        Login
+      </Button>
+    </div>
+  )
+}
+```
+
+- Vamos a intentarlo.
+
+```bash
+npm start
+```
+
+- Y vamos a añdir un alert (Ejercicio y una notificación) cuando el usuario pulse y los campos del formulario son válidos.
+
+_./src/pages/login/loginPageContainer.tsx_
+
+```diff
+  onLogin = () => {
++    loginFormValidation.validateForm(this.state.loginInfo)
++      .then((formValidationResult) => {
++          if(formValidationResult.succeeded) {
+            if (isValidLogin(this.state.loginInfo)) {
+              this.props.history.push('/pageB');
+            } else {
+              this.setState({ showLoginFailedMsg: true });
+            }      
++          } else {
++            alert('error, review the fields');
++          }
++      })
+  }
+```
+
+// TODO: mapFormValidationResultToFieldValidationErrors
```

Comment:
Remove this TODO, in english version as well

Reply:
Or rather (quite better), implement it by your self, ask for some guidance to @Nasdan